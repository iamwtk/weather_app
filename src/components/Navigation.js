import React from 'react'
import { NavLink } from 'react-router-dom'

import style from './navigation.scss';

const Navigation = (props) => (
  <div className={style.navigation}>
    <h1>Your best weather app</h1>
    <div className={style.separator}/>
    <ul>
      <li><NavLink onClick={props.closeNav} activeClassName={style.active} to='/about'>about</NavLink></li>
      <li><NavLink onClick={props.closeNav} activeClassName={style.active} to='/how-it-works'>how it works</NavLink></li>
      <li><NavLink onClick={props.toggleNav} activeClassName={style.active} to='/contact'>contact</NavLink></li>
    </ul>
  </div>
)

export default Navigation
