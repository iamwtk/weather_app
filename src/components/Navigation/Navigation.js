import React from 'react'
import { NavLink } from 'react-router-dom'

import style from './navigation.scss'

/**
 * <Navigation />
 * displays side navigation with link to sub pages
 * @param {function} closeNav dispatches action to close navigation on displays < 992px
 */
const Navigation = ({closeNav}) => {
  /**
   * Heading displayed in the upper part of application window
   * @type {String}
   */
  const heading = 'Your best weather app'
  /**
   * Array of pages and given paths
   * @type {Array}
   */
  const pages = [
    {text: 'about', path: '/about'},
    {text: 'how it works', path: '/how-it-works'},
    {text: 'contact', path: '/contact'}
  ]
  /**
   * Maps pages into NavLink elements
   * @type {ReactMarkup}
   */
  const links = pages.map((page, i) =>
    <li key={'page-'+i}><NavLink key={'link-'+i} onClick={closeNav} activeClassName={style.active} to={page.path}>{page.text}</NavLink></li>
  )
  /**
   * Displays markup of Navigation
   * @type {ReactMarkup}
   */
  return (
    <div className={style.navigation}>
      <h1>{heading}</h1>
      <div className={style.separator}/>
      <ul>
        {links}
      </ul>
    </div>
  )
}

export default Navigation
