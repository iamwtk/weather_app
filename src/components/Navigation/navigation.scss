@import "../../scss/variables";
@import "../../scss/animations";
@import "../../scss/mixins";

.navigation {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background: $purple;
  text-align: left;
  padding: 6.5em 1.5em 2em;
  animation: fadeIn .5s ease-in-out;

  @include md {
    width: 250px;
    animation: slideInLeft .5s ease-in-out;
  };
  @include lg {
    width: 300px;
    animation: slideInLeft .5s ease-in-out;
  };


  h1 {
    color: $white;
    display: table-caption;
    margin-bottom: 1.5em;
    font-family: $font-book;
    font-size: 2em;
    font-weight: 300;
  }
  .separator {
    display: inline-block;
    height:3px;
    background: $base;
    width: 55px;
  }

  ul {
    margin-top: 1em;
    li {
      a {
        display: block;
        font-family: $font-medium;
        font-size: 1.9em;
        color: $base;
        padding: .125em 0;
        text-decoration: none;
        transition-duration: .5s;

        &:hover {
          text-decoration: none;
          color: $white;
        }
        &.active {
          color: $white;
        }

      }
    }
  }

}
