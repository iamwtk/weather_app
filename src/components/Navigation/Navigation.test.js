import React from 'react'
import { shallow, mount } from 'enzyme'
import { NavLink } from 'react-router-dom';
import toJson from 'enzyme-to-json'

import Navigation from './Navigation'

import style from './navigation.scss'

describe('<Navigation />', () => {
  let component
  let i = 0
  function closeNav() {
    i++
  }
  beforeEach(() => {
    component = shallow(<Navigation closeNav={closeNav} />)
  })
  describe('Behaviour', () => {
    it('Should call closeNav function onclick on each NavLink(3) ', () => {
      const links = component.find(NavLink)
      links.map(link => {
        link.simulate('click')
      })      
      expect(i).toBe(3)
    })
  })
  describe('Snapshot', () => {
    it('Should match the Snapshot', () => {
      const tree = toJson(component)
      expect(tree).toMatchSnapshot()
    });
  });
})
