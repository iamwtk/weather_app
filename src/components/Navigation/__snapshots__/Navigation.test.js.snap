// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Snapshot Should match the Snapshot 1`] = `
<div
  className="navigation"
>
  <h1>
    Your best weather app
  </h1>
  <div
    className="separator"
  />
  <ul>
    <li
      key="page-0"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        key="link-0"
        onClick={[Function]}
        to="/about"
      >
        about
      </NavLink>
    </li>
    <li
      key="page-1"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        key="link-1"
        onClick={[Function]}
        to="/how-it-works"
      >
        how it works
      </NavLink>
    </li>
    <li
      key="page-2"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        key="link-2"
        onClick={[Function]}
        to="/contact"
      >
        contact
      </NavLink>
    </li>
  </ul>
</div>
`;
